crates/trigger/src/loader.rs

```diff
@@ -0,0 +1,126 @@
+use anyhow::Context as _;
+use spin_common::{ui::quoted_path, url::parse_file_url};
+use spin_core::{async_trait, wasmtime, Component};
+use spin_factors::AppComponent;
+
+#[derive(Default)]
+pub struct ComponentLoader {
+    #[cfg(feature = "unsafe-aot-compilation")]
+    aot_compilation_enabled: bool,
+}
+
+impl ComponentLoader {
+    /// Updates the TriggerLoader to load AOT precompiled components
+    ///
+    /// **Warning: This feature may bypass important security guarantees of the
+    /// Wasmtime security sandbox if used incorrectly! Read this documentation
+    /// carefully.**
+    ///
+    /// Usually, components are compiled just-in-time from portable Wasm
+    /// sources. This method causes components to instead be loaded
+    /// ahead-of-time as Wasmtime-precompiled native executable binaries.
+    /// Precompiled binaries must be produced with a compatible Wasmtime engine
+    /// using the same Wasmtime version and compiler target settings - typically
+    /// by a host with the same processor that will be executing them. See the
+    /// Wasmtime documentation for more information:
+    /// https://docs.rs/wasmtime/latest/wasmtime/struct.Module.html#method.deserialize
+    ///
+    /// # Safety
+    ///
+    /// This method is marked as `unsafe` because it enables potentially unsafe
+    /// behavior if used to load malformed or malicious precompiled binaries.
+    /// Loading sources from an incompatible Wasmtime engine will fail but is
+    /// otherwise safe. This method is safe if it can be guaranteed that
+    /// `<TriggerLoader as Loader>::load_component` will only ever be called
+    /// with a trusted `LockedComponentSource`. **Precompiled binaries must
+    /// never be loaded from untrusted sources.**
+    #[cfg(feature = "unsafe-aot-compilation")]
+    pub unsafe fn enable_loading_aot_compiled_components(&mut self) {
+        self.aot_compilation_enabled = true;
+    }
+
+    #[cfg(feature = "unsafe-aot-compilation")]
+    fn load_precompiled_component(
+        &self,
+        engine: &wasmtime::Engine,
+        path: &std::path::Path,
+    ) -> anyhow::Result<Component> {
+        assert!(self.aot_compilation_enabled);
```

Comment:
Is there a reason we're panicking here instead of using `anyhow::ensure!`?

Reply:
This is just a belt-and-suspenders regression assertion; this condition is unreachable with the current code.